File: c/gpt/p20.c
// p20: read value from command line and add constant OFFSET = 1000
#include <stdio.h>
#include <stdlib.h>

#define OFFSET 1000

int main(int argc, char **argv)
{
    if (argc < 2)
    {
        printf("Usage: %s <number>\n", argv[0]);
        return 1;
    }
    char *end;
    long x = strtol(argv[1], &end, 10);
    if (*end != '\0')
    {
        printf("Invalid number\n");
        return 1;
    }
    printf("%ld\n", x + OFFSET);
    return 0;
}

File: java/gpt/P20.java
public class P20 {
    private static final int OFFSET = 1000;
    public static void main(String[] args) {
        if (args.length < 1) {
            System.out.println("Usage: java P20 <number>");
            return;
        }
        try {
            int x = Integer.parseInt(args[0]);
            System.out.println(x + OFFSET);
        } catch (Exception e) {
            System.out.println("Invalid number");
        }
    }
}
